src/FileConventions/Library.fs

```diff
@@ -83,7 +88,9 @@ let DetectAsteriskInPackageReferenceItems(fileInfo: FileInfo) =
     asteriskInPackageReference.IsMatch fileText
 
 let DetectMissingVersionsInNugetPackageReferences(fileInfo: FileInfo) =
-    assert (fileInfo.FullName.EndsWith ".fsx")
+    Misc.BetterAssert
+        (fileInfo.FullName.EndsWith ".fsx")
+        sourceFileAssertionError
```

Comment:
@Mersho checking if the extension of the file is .fsx but later failing with error "Bug: file was not a F#/C# source file" would be extremely confusing because C# files have .cs extension, and F# files have .fs extension

Reply:
@knocte Thank you for reviewing. The library.fs has been edited. Please let me know if there is anything wrong.

Reply:
@Mersho if this has been addressed, please click on the button "Resolve conversation"